packages/test/end-to-end-tests/src/test/detachedContainerTests.spec.ts

```diff
@@ -242,6 +242,30 @@ const tests = (args: ILocalTestObjectProvider) => {
         assert.strictEqual(dataStore2.runtime.attachState, AttachState.Attached, "Data store 2 should be attached");
     });
 
+    it("Directly attach container through service factory, should resolve to same container", async () => {
+        const container = await loader.createDetachedContainer(pkg);
+        // Get the root dataStore from the detached container.
+        const response = await container.request({ url: "/" });
+        const dataStore = response.value as ITestFluidObject;
+
+        // Create a sub dataStore of type TestFluidObject.
+        const subDataStore1 = await createFluidObject(dataStore.context, "default");
+        dataStore.root.set("attachKey", subDataStore1.handle);
+
+        const snapshotTree = container.serialize();
+        const summaryForAttach = convertSnapshotTreeToProtocolAndAppSummaryTree(snapshotTree);
+        const resolvedUrl = await args.urlResolver.resolve(request);
+        const service = await args.documentServiceFactory.createContainer(summaryForAttach as any, resolvedUrl);
```

Comment:
why do we need to cast to any? It looks to be ISummaryTree  - both variable and argument are of same type

Reply:
They're incompatible because of the new SummaryType value as:
documentServiceFactory: IDocumentServiceFactory | old.IDocumentServiceFactory,
So needs to be casted to any.